lib/relish/encryption_helper.rb

```diff
@@ -35,9 +35,8 @@ def inspect
     protected
 
     def hmac_secrets
-      @hmac_secrets ||= @secrets.map do |secret|
-        OpenSSL::HMAC.hexdigest('sha256', @static_secret, secret)
-      end
+      @hmac_secrets ||= @static_secrets.product(@secrets).map {|static_secret, secret|
```

Comment:
can there be more than one @secrets per static_secrets? how does the product mapping work in that scenario?

Reply:
Yes, a static secret can have a current and a retired secret. Worst case scenario we can have a current and a retired static secret with a current and retired secret. 

The product functions makes all the combinations, see an example below
```
irb(main):004:0> static_secrets = ['static current', 'static retired']
=> ["static current", "static retired"]
irb(main):005:0> secrets = ['secret current', 'secret retired']
=> ["secret current", "secret retired"]
irb(main):006:0> static_secrets.product(secrets).map { |s, e| puts "#{s} - #{e}" }
static current - secret current
static current - secret retired
static retired - secret current
static retired - secret retired
=> [nil, nil, nil, nil]
```